src/components/MultiHopTrade/hooks/useSupportedAssets.tsx

```diff
@@ -0,0 +1,52 @@
+import { useMemo } from 'react'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { LifiSwapper } from 'lib/swapper/swappers/LifiSwapper/LifiSwapper'
+import { selectAssetsSortedByMarketCapFiatBalanceAndName } from 'state/slices/common-selectors'
+import { selectAssetIds, selectSellAsset } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+export const useSupportedAssets = () => {
+  const sellAsset = useAppSelector(selectSellAsset)
+  const assetIds = useAppSelector(selectAssetIds)
+  const sortedAssets = useAppSelector(selectAssetsSortedByMarketCapFiatBalanceAndName)
+  const isLifiEnabled = useFeatureFlag('LifiSwap')
+
+  const enabledSwappers = useMemo(() => {
+    const result = []
+    if (isLifiEnabled) result.push(LifiSwapper)
+    // TODO(woodenfurniture): add more swappers here
+    return result
+  }, [isLifiEnabled])
+
+  const supportedSellAssets = useMemo(() => {
```

Comment:
Isn't this whole function just going to leave us with the intersection of assets supported by all enabled swappers? 

It might currently work with only one swapper, lifi, but it doesn't feel like what we want.

Reply:
right you are, ill fix this

Reply:
my assumption is that we want the user to be able to sell or buy assets based on what's possible rather than what's selected? Otherwise if a user got a qwuote for say `inch`, they would only be able to buy/sell `1inch` supported assets until the refreshed the app

Reply:
ah pointed out by @0xApotheosis , we want the union not the intersection